docs/utilities/custom_resources.md

```diff
@@ -168,11 +204,89 @@ public class CustomSerializationHandler extends AbstractResourceHandler {
 
     @Override
     protected Response create(CloudFormationCustomResourceEvent createEvent, Context context) {
+        String physicalResourceId = "my-policy-name-" + UUID.randomUUID(); //Create a unique ID 
         Policy policy = new Policy();
         return Response.builder()
+                .status(Response.Status.SUCCESS)
+                .physicalResourceId(physicalResourceId)
                 .value(policy)
                 .objectMapper(policyMapper) // customize serialization
                 .build();
     }
 }
 ```
+
+## Advanced
+
+### Understanding the CloudFormation custom resource lifecycle
+
+While the library provides an easy-to-use interface, we recommend that you understand the lifecycle of CloudFormation custom resources before using them in production.
+
+#### Creating a custom resource
+When CloudFormation issues a CREATE on a custom resource, there are 2 possible states: `CREATE_COMPLETE` and `CREATE_FAILED`
+```mermaid
```

Comment:
This is sweet! I had no idea you could embed un-rendered diagrams into github markdown like this 🙌

Reply:
Do we know that it'll render out properly into the documentation website too?

Reply:
Yes it works, I validated locally following the guidance here https://docs.github.com/en/pages/setting-up-a-github-pages-site-with-jekyll/testing-your-github-pages-site-locally-with-jekyll?platform=mac